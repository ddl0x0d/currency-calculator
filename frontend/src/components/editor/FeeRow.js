import Button from 'react-bootstrap/Button';
import Col from 'react-bootstrap/Col';
import PropTypes from 'prop-types';
import React from 'react';
import Row from 'react-bootstrap/Row';

const propTypes = {
  fee: PropTypes.shape({
    id: PropTypes.string.isRequired,
    from: PropTypes.string.isRequired,
    to: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired
  }).isRequired,
  removeFee: PropTypes.func.isRequired
};

const FeeRow = props => {
  return (
    <Row className="justify-content-center align-items-baseline">
      <Col className="text-center" xs={3} md={2} xl={1}>
        {props.fee.from}
      </Col>
      <Col className="text-center" xs={3} md={2} xl={1}>
        {props.fee.to}
      </Col>
      <Col className="text-center" xs={3} md={2} xl={1}>
        {props.fee.amount.toFixed(2)}
      </Col>
      <Col xs={3} md={2} xl={1}>
        <Button
          className="mb-3"
          block="true"
          size="sm"
          variant="danger"
          onClick={() => props.removeFee(props.fee.id)}
        >
          Remove
        </Button>
      </Col>
    </Row>
  );
};

FeeRow.propTypes = propTypes;

export default FeeRow;
