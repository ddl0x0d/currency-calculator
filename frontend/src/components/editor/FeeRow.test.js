import Col from 'react-bootstrap/Col';
import FeeRow from './FeeRow';
import React from 'react';
import { shallow } from 'enzyme';

describe('<FeeRow />', () => {
  const removeFee = jest.fn().mockName('removeFee');
  const fee = { id: '123', from: 'EUR', to: 'GBP', amount: 0 };
  const feeRow = () => <FeeRow fee={fee} removeFee={removeFee} />;

  it('renders correctly', () => {
    const wrapper = shallow(feeRow());
    expect(wrapper).toMatchSnapshot();
  });

  it('renders fee amount with 2 decimal places', () => {
    const wrapper = shallow(feeRow());
    const amount = wrapper.find(Col).at(2);
    expect(amount.text()).toBe('0.00');
  });

  it('removes fee', () => {
    const wrapper = shallow(feeRow());
    wrapper.find('Button').simulate('click');
    expect(removeFee).toHaveBeenCalledWith('123');
  });
});
