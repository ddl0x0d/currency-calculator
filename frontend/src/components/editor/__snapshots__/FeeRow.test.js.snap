// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeeRow /> renders correctly 1`] = `
<Bootstrap(Row)
  className="justify-content-center align-items-baseline"
>
  <Col
    as="div"
    className="text-center"
    md={2}
    xl={1}
    xs={3}
  >
    EUR
  </Col>
  <Col
    as="div"
    className="text-center"
    md={2}
    xl={1}
    xs={3}
  >
    GBP
  </Col>
  <Col
    as="div"
    className="text-center"
    md={2}
    xl={1}
    xs={3}
  >
    0.00
  </Col>
  <Col
    as="div"
    md={2}
    xl={1}
    xs={3}
  >
    <Button
      active={false}
      block="true"
      className="mb-3"
      disabled={false}
      onClick={[Function]}
      size="sm"
      type="button"
      variant="danger"
    >
      Remove
    </Button>
  </Col>
</Bootstrap(Row)>
`;
